generators/upgrade/index.js

```diff
@@ -123,6 +123,16 @@ module.exports = class extends BaseGenerator {
 
     get configuring() {
         return {
+
+            assertJHipsterProject() {
+                const done = this.async();
+                if (!(shelljs.test('-f', '.yo-rc.json'))
+                    && !(shelljs.test('-d', '.jhipster'))) {
```

Comment:
the `.jhipster` directory will not be present if there are no entities I guess so probabaly just checking yo-rc is enough

Reply:
I thought that looking only at .yo-rc.json would match also non JHipster projects using yeoman.
Though we could have a valid JHipster project without entities if no database.

Reply:
You could check if the yo-rc object has `generator-jhipster` as a key

Reply:
Yes much better